File: test_dir/src/transpiler_config_test.py
from pypp_python import configclass, dataclass, Val


def a_function_i_like():
    print("I like functions!")


def a_function_i_like_extra():
    print("I like functions!")


@configclass
class _ConfigClassA:
    a: int = 10


@configclass
class _ConfigClassB:
    b: int = 20


@dataclass
class AnnAssignCustomType:
    x: Val[str]


@dataclass
class AnnAssignOtherType:
    x: Val[str]


def transpiler_config_test_fn():
    print("TRANSPILER CONFIG TEST:")
    crazy_name: str = "crazy"
    print(crazy_name)
    a_function_i_like()
    a: AnnAssignCustomType = AnnAssignCustomType("hello")
    print(a.x)
    print(_ConfigClassA.a)

File: test_dir/.pypp/cpp/src/transpiler_config_test.cpp
#include "src/transpiler_config_test.h"
#include "pypp_util/print.h"
#include <string>
#include <vector>

namespace me {
void a_function_i_like() { pypp::print(pypp::PyStr("I like functions!")); }

void a_function_i_like_extra() {
    pypp::print(pypp::PyStr("I like functions!"));
}

struct __PseudoPyppName_ConfigClassA {
    int a = 10;
};
inline __PseudoPyppName_ConfigClassA _ConfigClassA;

struct __PseudoPyppName_ConfigClassB {
    int b = 20;
};
inline __PseudoPyppName_ConfigClassB _ConfigClassB;

void transpiler_config_test_fn() {
    pypp::print(pypp::PyStr("TRANSPILER CONFIG TEST:"));
    pypp::PyStr crazy_name_extra = pypp::PyStr("crazy");
    pypp::print(crazy_name_extra);
    a_function_i_like_extra();
    AnnAssignOtherType a = AnnAssignOtherType(pypp::PyStr("hello"));
    pypp::print(a.x);
    pypp::print(_ConfigClassB.b);
}

} // namespace me